azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -36,6 +43,14 @@ class AsyncAzureCliCredential:
         def __init__(self):
             raise KustoAioSyntaxError()
 
+    class AsyncDefaultAzureCredential:
+        def __init__(self):
+            raise KustoAioSyntaxError()
+
+    class AsyncTokenCredential:
+        def __init__(self):
+            raise KustoAioSyntaxError()

```

Comment:
This is in case that the optional async module is uninstalled.

This gives a good error for the user if they try to use async types without the module.